src/main/java/com/jhuguet/sb_taskv1/app/handler/GlobalExceptionHandler.java

```diff
@@ -108,4 +111,22 @@ public ResponseEntity<?> wrongSortOrder(WrongSortOrder exception) {
         return new ResponseEntity<>(details, HttpStatus.BAD_REQUEST);
     }
 
+    @ExceptionHandler(WrongCredentials.class)
+    public ResponseEntity<?> wrongCredentials(WrongCredentials exception) {
+        ExceptionDetails details = new ExceptionDetails(exception.getMessage());
+        return new ResponseEntity<>(details, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(UnqualifiedAuthority.class)
+    public ResponseEntity<?> unqualifiedAuthority(UnqualifiedAuthority exception) {
+        ExceptionDetails details = new ExceptionDetails(exception.getMessage());
+        return new ResponseEntity<>(details, HttpStatus.FORBIDDEN);
+    }
+
+    @ExceptionHandler(UsernameNotFound.class)
+    public ResponseEntity<?> usernameNotFound(UsernameNotFound exception) {
```

Comment:
I think, that 401 code (UNAUTHORIZED) here is better. Check the article https://auth0.com/blog/forbidden-unauthorized-http-status-codes/

Reply:
According to the article, shouldn't we use 404 to avoid letting the client know information that could potentially harm the account holder?

Reply:
Good point, but community from stackoverflow use 401 in this way. https://stackoverflow.com/questions/32752578/whats-the-appropriate-http-status-code-to-return-if-a-user-tries-logging-in-wit
Let's discuss on the meeting.